bundles/org.openhab.binding.livisismarthome/src/main/java/org/openhab/binding/livisismarthome/internal/handler/LivisiDeviceHandler.java

```diff
@@ -0,0 +1,1003 @@
+            case CapabilityDTO.TYPE_WINDOWDOORSENSOR:
+                updateWindowDoorSensorChannels(capability);
+                break;
+            case CapabilityDTO.TYPE_SMOKEDETECTORSENSOR:
+                updateSmokeDetectorChannels(capability);
+                break;
+            case CapabilityDTO.TYPE_ALARMACTUATOR:
+                updateAlarmActuatorChannels(capability);
+                break;
+            case CapabilityDTO.TYPE_MOTIONDETECTIONSENSOR:
+                updateMotionDetectionSensorChannels(capability);
+                break;
+            case CapabilityDTO.TYPE_LUMINANCESENSOR:
+                updateLuminanceSensorChannels(capability);
+                break;
+            case CapabilityDTO.TYPE_PUSHBUTTONSENSOR:
+                updatePushButtonSensorChannels(capability);
+                break;
+            case CapabilityDTO.TYPE_ENERGYCONSUMPTIONSENSOR:
+                updateEnergyConsumptionSensorChannels(capability);
+                break;
+            case CapabilityDTO.TYPE_POWERCONSUMPTIONSENSOR:
+                updateStateForEnergyChannelWatt(CHANNEL_POWER_CONSUMPTION_WATT,
+                        capability.getCapabilityState().getPowerConsumptionSensorPowerConsumptionWattState(),
+                        capability);
+                break;
+            case CapabilityDTO.TYPE_GENERATIONMETERENERGYSENSOR:
+                updateGenerationMeterEnergySensorChannels(capability);
+                break;
+            case CapabilityDTO.TYPE_GENERATIONMETERPOWERCONSUMPTIONSENSOR:
+                updateStateForEnergyChannelWatt(CHANNEL_POWER_GENERATION_WATT,
+                        capability.getCapabilityState().getGenerationMeterPowerConsumptionSensorPowerInWattState(),
+                        capability);
+                break;
+            case CapabilityDTO.TYPE_TWOWAYMETERENERGYCONSUMPTIONSENSOR:
+                updateTwoWayMeterEnergyConsumptionSensorChannels(capability);
+                break;
+            case CapabilityDTO.TYPE_TWOWAYMETERENERGYFEEDSENSOR:
+                updateTwoWayMeterEnergyFeedSensorChannels(capability);
+                break;
+            case CapabilityDTO.TYPE_TWOWAYMETERPOWERCONSUMPTIONSENSOR:
+                updateStateForEnergyChannelWatt(CHANNEL_POWER_WATT,
+                        capability.getCapabilityState().getTwoWayMeterPowerConsumptionSensorPowerInWattState(),
+                        capability);
+                break;
+            default:
+                logger.debug("Unsupported capability type {}.", capability.getType());
+                break;
+        }
+    }
+
+    private void updateVariableActuatorChannels(CapabilityDTO capability) {
+        final Boolean variableActuatorState = capability.getCapabilityState().getVariableActuatorState();
+        if (variableActuatorState != null) {
+            updateState(CHANNEL_SWITCH, OnOffType.from(variableActuatorState));
+        } else {
+            logStateNULL(capability);
```

Comment:
Perhaps channel should be updated to **UNDEF** in this case. The same applies to all other methods.

Reply:
Hm, I think we shouldn't change that. The innogysmarthome binding is used for many years by many users and there are no known bugs. The device data should be received completely, so this can't happen. There are websocket events which doesn't contain all device data, but in this case a missing attribute doesn't mean that it is missing / undefined instead it means that there is no new data.

Reply:
The innogysmarthome binding is available since several years and there are no known bugs, so I think here isn't a bug. I would not like to change that. And I think when it should happen that an attribute is missing, it means that there is no change / the old value is still valid. In this case it would be wrong to set the channel to undefined. This happens for sure on receiving events, events doesn't contain all attributes of the device (again).